compiler/rustc_interface/src/passes.rs

```diff
@@ -952,6 +958,151 @@ fn analysis(tcx: TyCtxt<'_>, (): ()) -> Result<()> {
     Ok(())
 }
 
+fn non_par_analysis(tcx: TyCtxt<'_>) -> Result<()> {
```

Comment:
This code duplication is really unfortunate.

Reply:
That's right. Haven't thought of an elegant way to write it yet, but I'll fix that soon